src/EnergyPlus/PurchasedAirManager.cc

```diff
@@ -233,876 +233,953 @@ void GetPurchasedAir(EnergyPlusData &state)
+                if (PurchAir.DCVType == DCV::CO2SetPoint) {
+                    if (!state.dataContaminantBalance->Contaminant.CO2Simulation) {
+                        PurchAir.DCVType = DCV::None;
+                        ShowWarningError(state, format("{}{}=\"{} invalid data", RoutineName, s_ipsc->cCurrentModuleObject, PurchAir.Name));
+                        ShowContinueError(state, format("{}={} but CO2 simulation is not active.", cAlphaFieldName, fieldValue));
+                        ShowContinueError(state, format("Resetting {} to NoDCV", cAlphaFieldName));
+                        ShowContinueError(state,
+                                          "To activate CO2 simulation, use ZoneAirContaminantBalance object and specify \"Carbon Dioxide "
+                                          "Concentration\"=\"Yes\".");
+                    }
+                } else if (PurchAir.DCVType == DCV::Invalid) {
+                    ShowSevereInvalidKey(state, eoh, cAlphaFieldName, fieldValue, "Valid entries are None, OccupancySchedule, or CO2Setpoint.");
                     ErrorsFound = true;
-                } else if (PurchAir(PurchAirNum).DCVType == DCV::CO2SetPoint && !state.dataContaminantBalance->Contaminant.CO2Simulation) {
-                    PurchAir(PurchAirNum).DCVType = DCV::None;
-                    ShowWarningError(state, format("{}{}=\"{} invalid data", RoutineName, s_ipsc->cCurrentModuleObject, s_ipsc->cAlphaArgs(1)));
-                    ShowContinueError(state, format("{}={} but CO2 simulation is not active.", s_ipsc->cAlphaFieldNames(14), s_ipsc->cAlphaArgs(14)));
-                    ShowContinueError(state, format("Resetting {} to NoDCV", s_ipsc->cAlphaFieldNames(14)));
-                    ShowContinueError(state,
-                                      "To activate CO2 simulation, use ZoneAirContaminantBalance object and specify \"Carbon Dioxide "
-                                      "Concentration\"=\"Yes\".");
                 }
 
                 // get Outdoor air economizer type
-                PurchAir(PurchAirNum).EconomizerType = static_cast<Econ>(getEnumValue(econNamesUC, s_ipsc->cAlphaArgs(15)));
-                if (PurchAir(PurchAirNum).EconomizerType == Econ::Invalid) {
-                    ShowSevereInvalidKey(state,
-                                         eoh,
-                                         s_ipsc->cAlphaFieldNames(15),
-                                         s_ipsc->cAlphaArgs(15),
-                                         "Valid entries are NoEconomizer, DifferentialDryBulb, or DifferentialEnthalpy");
+                cAlphaFieldName = "Outdoor Air Economizer Type";
+                fieldValue = s_ip->getAlphaFieldValue(fields, schemaProps, "outdoor_air_economizer_type");
+                PurchAir.EconomizerType = static_cast<Econ>(getEnumValue(econNamesUC, Util::makeUPPER(fieldValue)));
+                if (PurchAir.EconomizerType == Econ::Invalid) {
```

Comment:
Not possible, delete.

    A15, \field Outdoor Air Economizer Type
       \type choice
       \key NoEconomizer
       \key DifferentialDryBulb
       \key DifferentialEnthalpy
       \default NoEconomizer